src/arcaflow_plugin_sdk/atp.py

```diff
@@ -74,13 +74,13 @@ class ATPServer:
     input_pipe: io.FileIO
     output_pipe: io.FileIO
     stderr: io.FileIO
-    step_ids: typing.Dict[str, str] = {} # Run ID to step IDs
-    encoder: cbor2.encoder
-    decoder: cbor2.decoder
+    step_ids: typing.Dict[str, str]  # Run ID to step IDs
+    encoder: cbor2.CBOREncoder
+    decoder: cbor2.CBOREncoder
```

Comment:
I think this should be `CBORDecoder`??

However, this whole section is "interesting" ... you're declaring class attributes, but below you're using them as instance attributes. These are only assigning type hints, not allocating storage or anything, and I presume the intent is purely as documentation: but this also implies the use of these names as class attributes, which is misleading. (And of course in Python there's no rule that the type of the actual instance attributes, even with the same name, need to match: they're completely different attributes.) This feels very much like "someone" was trying to pretend Python was strongly/statically typed, and that suit doesn't really fit well. 😆 

It'd be better documentation (and coding practice) to annotate the `__init__` assignments, instead, if you want type hints. (Then again, I recognize that this seems to be a general practice throughout the module, and Python is very much a "do what thou wilt" language, so take this with a grain or two of salt.)

Reply:
Good catch. I'll have that changed to Decoder in the next commit.

The modern type annotations have the instance variables set as just types of unassigned class attributes that are then assigned in the constructor. For class variables, it's the same except your supposed to add ClassVar to it.
https://peps.python.org/pep-0526/#class-and-instance-variable-annotations

Clearly these don't do much in Python. As shown by there being no error with the decoder being assigned to the decoder.